trappy/ftrace.py

```diff
@@ -79,6 +120,22 @@ def __init__(self, name="", normalize_time=True, scope="all",
             setattr(self, attr, trace_class)
             self.trace_classes.append(trace_class)
 
+        cache_good = False
+
+        if not self.__class__.disable_cache and self._check_trace_cache():
+            cache_good = True
+            try:
+                # Read csv into frames
+                for trace_class in self.trace_classes:
+                    csv_file = self._get_csv_path(trace_class)
+                    trace_class.read_csv(csv_file)
+            except:
+                warnings.warn("TRAPpy: Caching disabled due to cache parsing error")
+                cache_good = False
```

Comment:
You can still get rid of `cache_good` ([previous comment here](https://github.com/ARM-software/trappy/pull/247#discussion_r116565256))